gk/main.c

```diff
@@ -552,6 +552,72 @@ gk_setup_rss(struct gk_config *gk_conf)
 	return ret;
 }
 
+static void
+gk_arp_req_cb(const struct lls_map *map, void *arg,
+	__attribute__((unused))enum lls_reply_ty ty, int *pcall_again)
+{
+	unsigned int i;
+	struct gk_config *gk_conf = arg;
+	struct gk_lpm *ltbl = gk_conf->lpm_tbl;
+
+	for (i = 0; i < ltbl->num_tunnels; i++) {
+		if (ltbl->tunnels[i].flow.proto == ETHER_TYPE_IPv4
+				&& memcmp(&ltbl->tunnels[i].flow.f.v6.dst,
+				map->ip_be, sizeof(struct in_addr)) == 0) {
+			ether_addr_copy(&map->ha,
+				&ltbl->tunnels[i].nexthop_mac);
+			break;
+		}
+	}
+
+	*pcall_again = true;
+}
+
+static void
+gk_nd_req_cb(const struct lls_map *map, void *arg,
+	__attribute__((unused))enum lls_reply_ty ty, int *pcall_again)
+{
+	unsigned int i;
+	struct gk_config *gk_conf = arg;
+	struct gk_lpm *ltbl = gk_conf->lpm_tbl;
+
+	for (i = 0; i < ltbl->num_tunnels; i++) {
+		if (ltbl->tunnels[i].flow.proto == ETHER_TYPE_IPv6
```

Comment:
Thanks! That's true. I just ignored the fact that ``hold_arp()`` or ``hold_nd()`` only needs the ``arg`` related to the specific IP address specified by the third parameter.